fv3core/stencils/c_sw.py

```diff
@@ -311,174 +302,285 @@ def update_meridional_velocity(
+        self._update_y_velocity = FrozenStencil(
+            func=update_y_velocity,
+            externals={
+                "grid_type": grid_type,
+                **axis_offsets_meridional,
+            },
+            origin=origin,
+            domain=domain_meridional,
+        )
+        domain_zonal = self.grid.domain_shape_compute(add=(1, 0, 0))
+        axis_offsets_zonal = axis_offsets(self.grid, origin, domain_zonal)
+        self._update_x_velocity = FrozenStencil(
+            func=update_x_velocity,
+            externals={
+                "grid_type": grid_type,
+                **axis_offsets_zonal,
+            },
+            origin=origin,
+            domain=domain_zonal,
+        )
+
+    def _vorticitytransport_cgrid(
+        self,
+        uc: FloatField,
+        vc: FloatField,
+        vort_c: FloatField,
+        ke_c: FloatField,
+        v: FloatField,
+        u: FloatField,
+        dt2: float,
+    ):
+        """Update the C-Grid x and y velocity fields.
+
+        Args:
+            uc: x-velocity on C-grid (input, output)
+            vc: y-velocity on C-grid (input, output)
+            vort_c: Vorticity on C-grid (input)
+            ke_c: kinetic energy on C-grid (input)
+            v: y-velocity on D-grid (input)
+            u: x-velocity on D-grid (input)
+            dt2: timestep (input)
+        """
+        self._update_y_velocity(
+            vort_c,
+            ke_c,
             u,
+            vc,
+            self.grid.cosa_v,
+            self.grid.sina_v,
+            self.grid.rdyc,
+            dt2,
+        )
+        self._update_x_velocity(
+            vort_c,
+            ke_c,
             v,
+            uc,
+            self.grid.cosa_u,
+            self.grid.sina_u,
+            self.grid.rdxc,
+            dt2,
+        )
+
+    def __call__(
+        self,
+        delp: FloatField,
+        pt: FloatField,
+        u: FloatField,
+        v: FloatField,
+        w: FloatField,
+        uc: FloatField,
+        vc: FloatField,
+        ua: FloatField,
+        va: FloatField,
+        ut: FloatField,
+        vt: FloatField,
+        divgd: FloatField,
+        omga: FloatField,
+        dt2: float,
+    ):
+        """
+        C-grid shallow water routine.
+
+        Advances C-grid winds by half a time step.
+        Args:
+            delp: D-grid vertical delta in pressure (in)
+            pt: D-grid potential temperature (in)
+            u: D-grid x-velocity (in)
+            v: D-grid y-velocity (in)
+            w: vertical velocity (in)
+            uc: C-grid x-velocity (inout)
+            vc: C-grid y-velocity (inout)
+            ua: A-grid x-velocity (in)
+            va: A-grid y-velocity (in)
+            ut: u * dx (inout)
+            vt: v * dy (inout)
+            divgd: D-grid horizontal divergence (inout)
+            omga: Vertical pressure velocity (inout)
+            dt2: Acoustic timestep in seconds (in)
```

Comment:
I believe `dt2` is half of the acoustic timestep. In other words, it's the timestep that this routine evolves the winds over, since it gets the winds halfway through the timestep. You may want to dig into this and update the name/description.